python/lsst/meas/astrom/astrometry_net.cc

```diff
@@ -0,0 +1,598 @@
+        std::shared_ptr<afw::coord::Coord const> crval
+            (new afw::coord::Coord(wcs->crval[0] * afw::geom::degrees,
+                                 wcs->crval[1] * afw::geom::degrees));
+        return afw::image::makeWcs(*crval, crpix,
+                                 wcs->cd[0][0], wcs->cd[0][1],
+                                 wcs->cd[1][0], wcs->cd[1][1]);
+    }
+
+    bool didSolve() {
+        return solver_did_solve(_solver.get());
+    }
+
+    void run(double cpulimit) {
+        solver_log_params(_solver.get());
+        struct timer_baton tt;
+        if (cpulimit > 0.) {
+            tt.s = _solver.get();
+            tt.timelimit = cpulimit;
+            _solver->userdata = &tt;
+            _solver->timer_callback = timer_callback;
+        }
+        solver_run(_solver.get());
+        if (cpulimit > 0.) {
+            _solver->timer_callback = NULL;
+            _solver->userdata = NULL;
+        }
+    }
+
+    double getQuadSizeLow() {
+        double qlo,qhi;
+        solver_get_quad_size_range_arcsec(_solver.get(), &qlo, &qhi);
+        return qlo;
+    }
+    double getQuadSizeHigh() {
+        double qlo,qhi;
+        solver_get_quad_size_range_arcsec(_solver.get(), &qlo, &qhi);
+        return qhi;
+    }
+
+    /**
+     * Add indices to the solver
+     *
+     * The indices are bare pointers whose memory is managed by the caller.
+     * Typically the indices are owned by a MultiIndex object owned by the caller.
+     */
+    void addIndices(std::vector<index_t*> inds) {
+        for (std::vector<index_t*>::iterator pind = inds.begin();
+             pind != inds.end(); ++pind) {
+            lsst::meas::astrom::detail::IndexManager man(*pind);
+//            printf("Checking index \"%s\"\n", man.index->indexname);
+            if (_solver->use_radec) {
+                double ra,dec,radius;
+                xyzarr2radecdeg(_solver->centerxyz, &ra, &dec);
+                radius = distsq2deg(_solver->r2);
+                if (!index_is_within_range(man.index, ra, dec, radius)) {
+                                     //printf("Not within RA,Dec range\n");
+                    continue;
+                }
+            }
+            // qlo,qhi in arcsec
+            double qlo, qhi;
+            solver_get_quad_size_range_arcsec(_solver.get(), &qlo, &qhi);
+            if (!index_overlaps_scale_range(man.index, qlo, qhi)) {
+//                printf("Not within quad scale range\n");
+                continue;
+            }
+//            printf("Adding index.\n");
+            if (index_reload(man.index)) {
+                throw LSST_EXCEPT(lsst::pex::exceptions::IoError,
+                                  "Failed to index_reload() an astrometry_net_data index file -- out of file descriptors?");
+            }
+
+            solver_add_index(_solver.get(), man.index);
+        }
+    }
+
+    /**
+     * Set parity to flipped (if true) or normal (if false)
+     */
+    void setParity(bool p) {
```

Comment:
The current name matches the `astometry_net` call so I'm doubly inclined to leave it alone. However, I will rename the argument `flipped`, which I think will help.